fairroot/base/sim/FairMCApplication.cxx

```diff
@@ -868,16 +871,9 @@ void FairMCApplication::InitGeometry()
     /// save Geo Params in Output file
     if (fRootManager && !fParent) {
         RegisterOutput();
-
-        /**Tasks has to be initialized here, they have access to the detector branches and still can create objects in
-         * the tree*/
-        /// There is always a Main Task  !
-        /// so .. always a InitTasks() is called <D.B>
-        if (fFairTaskList) {
-            InitTasks();
+        if (GetIsMT()) {
+            fRootManager->WriteFolder();
```

Comment:
The simplest answer is: it is the only combination that was working.

The long one: in the initialization of `VMC` users register their branches and the information about branch names and addresses is stored in a specific `TFolder` in `gROOT`. The function `WriteFolder` creates the output tree based on that very `TFolder`. In the `MTMode` all this initialization is done twice, once on the parent, second time on the worker. In order to store and use proper branch addresses on the worker, the parent `TFolder` has to be removed from `gROOT` before the new `TFolder` is created on the worker. 

**The function `WriteFolder` (since the introduction of the MTMode) was misused to remove the `TFolder` from `gROOT` after it was used to create the output tree and purely for that reason `WriteFolder` was called in the `InitGeometry`.**

**I can see now that it is confusing, so I will add a function `FairRootManager::RemoveFolder()` which will purely remove the `TFolder` and call it from `InitGeometry`.**

In the `InitGeometry` I do not need to guard mutex, since here it is only called on the parent.